cvmfs/object_fetcher.h

```diff
@@ -0,0 +1,387 @@
+/**
+ * This file is part of the CernVM File System.
+ */
+
+#ifndef OBJECT_FETCHER_H
+#define OBJECT_FETCHER_H
+
+#include <string>
+
+#include "manifest.h"
+#include "manifest_fetch.h"
+#include "history_sqlite.h"
+#include "catalog.h"
+
```

Comment:
At the moment the order is random so we can just start to introduce an order for all new things.